### PR TITLE
Fixes and additional examples to Gen2 Custom Resources Docs

### PR DESCRIPTION
**Pull request by kevinold**
#### Description of changes:

Fixes and additional examples to Gen2 Custom Resources Docs

#### Related GitHub issue #, if available:

[● JTBD Guides and Samsara Docs KANBAN - vNext Guides - Add any AWS Svc - Custom Resources - Asana](https://app.asana.com/0/1205212189033571/1205381356323141)

### Instructions

**If this PR should not be merged upon approval for any reason, please submit as a DRAFT**

Which product(s) are affected by this PR (if applicable)?
- [ ] amplify-cli
- [ ] amplify-ui
- [ ] amplify-studio
- [ ] amplify-hosting
- [ ] amplify-libraries

Which platform(s) are affected by this PR (if applicable)?
- [ ] JS
- [ ] iOS
- [ ] Android
- [ ] Flutter
- [ ] React Native

**Please add the product(s)/platform(s) affected to the PR title**

#### Checks

- [ ] Does this PR conform to [the styleguide](https://github.com/aws-amplify/docs/blob/main/STYLEGUIDE.md)?

- [ ] Does this PR include filetypes other than markdown or images? Please add or update unit tests accordingly.

- [ ] Are any files being deleted with this PR? If so, have the needed redirects been created?

- [ ] Are all links in MDX files using the MDX link syntax rather than HTML link syntax? <br /> 
      _ref: MDX: `[link](https://link.com)` 
            HTML: `<a href="https://link.com">link</a>`_

### When this PR is ready to merge, please check the box below
- [x] Ready to merge

_By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license._

### CODE REVIEW

File: src/pages/gen2/build-a-backend/add-aws-services/custom-resources/index.mdx [hunk 1]
```diff
@@ -29,71 +33,195 @@ Custom resources can be written using the [AWS Cloud Development Kit (CDK)](http
...
+
+    const { sourceAddress } = props;
+
+    // Create SNS topic
+    const topic = new sns.Topic(this, "NotificationTopic");
+
+    const nodeJsFunctionProps: NodejsFunctionProps = {
+      bundling: {
+        externalModules: [
+          "aws-sdk", // Use the 'aws-sdk' available in the Lambda runtime
```
Comment (josefaidt) [inline src/pages/gen2/build-a-backend/add-aws-services/custom-resources/index.mdx:89]: The Node 18 runtime comes with AWS SDK v3, so it should be safe to remove this

File: src/pages/gen2/build-a-backend/add-aws-services/custom-resources/index.mdx [hunk 2]
```diff
@@ -13,13 +13,17 @@ export function getStaticProps(context) {
...
----
-title: Custom Resources
----
+Custom resources can be written using the [AWS Cloud Development Kit (CDK)](https://docs.aws.amazon.com/cdk/api/v2/).
 
-@TODO better examples of constructs and stacks
+From the [Concepts - AWS Cloud Development Kit (AWS CDK) v2](https://docs.aws.amazon.com/cdk/v2/guide/core_concepts.html) documentation
 
-Custom resources can be written using the [AWS Cloud Development Kit (CDK)](https://docs.aws.amazon.com/cdk/api/v2/). You have the option to create individual [Constructs](https://docs.aws.amazon.com/cdk/v2/guide/constructs.html) or [Stacks](https://docs.aws.amazon.com/cdk/v2/guide/stacks.html).
+:::info
```
Comment (josefaidt) [inline src/pages/gen2/build-a-backend/add-aws-services/custom-resources/index.mdx:20]: this (and other references like this) are from [docusaurus's admonitions](https://docusaurus.io/docs/markdown-features/admonitions), whic unfortunately do not exist in this docs site. Can we port this to a `<Callout>`?

File: src/pages/gen2/build-a-backend/add-aws-services/custom-resources/index.mdx [hunk 3]
```diff
@@ -29,71 +33,195 @@ Custom resources can be written using the [AWS Cloud Development Kit (CDK)](http
...
+    throw err;
+  }
+};
+
+```
+
+```ts title="amplify/custom/CustomNotifications/emailer.ts"
+// emailer Lambda
+import { SNSEvent } from "@types/aws-lambda";
+import { SNSHandler } from "aws-lambda";
```
Comment (josefaidt) [inline src/pages/gen2/build-a-backend/add-aws-services/custom-resources/index.mdx:155]:
```suggestion
import type { SNSHandler } from "aws-lambda";
```

File: src/pages/gen2/build-a-backend/add-aws-services/custom-resources/index.mdx [hunk 4]
```diff
@@ -29,71 +33,195 @@ Custom resources can be written using the [AWS Cloud Development Kit (CDK)](http
...
+import { SES, SNS } from "aws-sdk";
```
Comment (josefaidt) [inline src/pages/gen2/build-a-backend/add-aws-services/custom-resources/index.mdx:156]: we should use AWS SDK v3 by default

File: src/pages/gen2/build-a-backend/add-aws-services/custom-resources/index.mdx [hunk 5]
```diff
@@ -29,71 +33,195 @@ Custom resources can be written using the [AWS Cloud Development Kit (CDK)](http
...
+import { Message } from "./resource";
+
+const topic = new SNS();
+const ses = new SES();
+
+export const handler: SNSHandler = async (event: SNSEvent) => {
```
Comment (josefaidt) [inline src/pages/gen2/build-a-backend/add-aws-services/custom-resources/index.mdx:162]:
```suggestion
export const handler: SNSHandler = async (event) => {
```

the event should be typed from `SNSHandler`

File: src/pages/gen2/build-a-backend/add-aws-services/custom-resources/index.mdx [hunk 6]
```diff
@@ -29,71 +33,195 @@ Custom resources can be written using the [AWS Cloud Development Kit (CDK)](http
...
 
 ```ts title="amplify/backend.ts"
-import { Backend } from '@aws-amplify/backend';
-import { Backup } from './custom/Backup';
+import { defineBackend } from '@aws-amplify/backend';
+import { CustomNotifications } from './custom/CustomNotifications/resource';
 import { data } from './data/resource';
 
-const backend = new Backend({
+const backend = new defineBackend({
```
Comment (josefaidt) [inline src/pages/gen2/build-a-backend/add-aws-services/custom-resources/index.mdx:216]:
```suggestion
const backend = defineBackend({
```

File: src/pages/gen2/build-a-backend/add-aws-services/custom-resources/index.mdx [hunk 7]
```diff
@@ -110,69 +238,140 @@ Resources created by the `define*` helpers can only be accessed _after_ setting
...
+    // Create a security group to be used on the lambda functions
+    const lambdaSecurityGroup = new ec2.SecurityGroup(
+      this,
+      "Lambda Security Group",
+      {
+        vpc,
+      }
+    );
+
+    const getDataLambda: NodejsFunction = new NodejsFunction(
```
Comment (josefaidt) [inline src/pages/gen2/build-a-backend/add-aws-services/custom-resources/index.mdx:291]:
```suggestion
    const getDataLambda = new NodejsFunction(
```

type will be inferred

File: src/pages/gen2/build-a-backend/add-aws-services/custom-resources/index.mdx [hunk 8]
```diff
@@ -110,69 +238,140 @@ Resources created by the `define*` helpers can only be accessed _after_ setting
...
+      this,
+      id + "-getDataLambda",
```
Comment (josefaidt) [inline src/pages/gen2/build-a-backend/add-aws-services/custom-resources/index.mdx:293]:
```suggestion
      `${id}-getDataLambda`,
```

File: src/pages/gen2/build-a-backend/add-aws-services/custom-resources/index.mdx [hunk 9]
```diff
@@ -110,69 +238,140 @@ Resources created by the `define*` helpers can only be accessed _after_ setting
...
+    states: [
+      "Alabama",
+      "Alaska",
+      "Arizona",
+      //...
+    ],
+  },
+];
+
+export async function handler(): Promise<APIGatewayProxyResultV2> {
```
Comment (josefaidt) [inline src/pages/gen2/build-a-backend/add-aws-services/custom-resources/index.mdx:332]: this signature is very different from the previous example, is this intentional? https://github.com/aws-amplify/docs/pull/6265/files#diff-41d0a6d7329d8852de45154989ef0af799ddbac5e7f825ec335e49fe5f31882eR132

File: src/pages/gen2/build-a-backend/add-aws-services/custom-resources/index.mdx [hunk 10]
```diff
@@ -110,69 +238,140 @@ Resources created by the `define*` helpers can only be accessed _after_ setting
...
+      exportName: `${id}-getDataLambdaArn`,
     });
   }
 }
 ```
 
+The Lambda function code for the `getDataLambda` is:
+
+```ts title="amplify/custom/LambdaInAVpcStack/getData.ts"
+import { APIGatewayProxyResultV2 } from "aws-lambda";
```
Comment (josefaidt) [inline src/pages/gen2/build-a-backend/add-aws-services/custom-resources/index.mdx:317]:
```suggestion
import type { APIGatewayProxyResultV2 } from "aws-lambda";
```

File: src/pages/gen2/build-a-backend/add-aws-services/custom-resources/index.mdx [hunk 11]
```diff
@@ -110,69 +238,140 @@ Resources created by the `define*` helpers can only be accessed _after_ setting
...
+
 ```ts title="amplify/backend.ts"
-import { Backend } from '@aws-amplify/backend';
-import { BackupStack } from './custom/BackupStack';
+import { defineBackend } from '@aws-amplify/backend';
+import { CustomNotifications } from './custom/CustomNotifications/resource';
 import { data } from './data/resource';
 
-const backend = new Backend({
+const backend = new defineBackend({
```
Comment (josefaidt) [inline src/pages/gen2/build-a-backend/add-aws-services/custom-resources/index.mdx:359]:
```suggestion
const backend = defineBackend({
```

File: src/pages/gen2/build-a-backend/add-aws-services/custom-resources/index.mdx [hunk 12]
```diff
@@ -29,71 +33,195 @@ Custom resources can be written using the [AWS Cloud Development Kit (CDK)](http
 
 :::
```
Comment (josefaidt) [inline src/pages/gen2/build-a-backend/add-aws-services/custom-resources/index.mdx:34]: same note about admonitions

File: src/pages/gen2/build-a-backend/add-aws-services/custom-resources/index.mdx (hunk 1 above)
Comment (kevinold) [reply src/pages/gen2/build-a-backend/add-aws-services/custom-resources/index.mdx:93]: I needed this when deploying it in my sandbox, so leaving it for now.

**Pull request merged**